controllers/capa_controller.go

```diff
@@ -79,14 +80,23 @@ func (r *CAPAClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, microerror.Mask(client.IgnoreNotFound(err))
 	}
 
+	if annotations.HasPaused(awsCluster) {
+		logger.Info("AWSCluster is marked as paused, skipping")
+		return ctrl.Result{}, nil
+	}
+
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, awsCluster.ObjectMeta)
 	if err != nil {
 		return reconcile.Result{}, microerror.Mask(err)
 	}
 
-	if annotations.HasPaused(awsCluster) {
-		logger.Info("AWSCluster is marked as paused, skipping")
-		return ctrl.Result{}, nil
+	// If the cluster is already deleted the configmap and role identity will
```

Comment:
The role identity is shared among different WCs, right? I don't think it's removed when removing a cluster.

Reply:
yeah you're right. I guess I was thinking of the acceptance tests in resolver-rules-operator where it's removed after each test